sdk/src/Core/Amazon.Runtime/Telemetry/Attributes.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ * 
+ *  http://aws.amazon.com/apache2.0
+ * 
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+using System.Collections.Generic;
+
+namespace Amazon.Runtime.Telemetry
+{
+    /// <summary>
+    /// Represents a collection of key-value pairs for attributes.
+    /// </summary>
+    public class Attributes
+    {
+        private readonly List<KeyValuePair<string, object>> _attributes;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Attributes"/> class.
+        /// </summary>
+        /// <param name="attributes">An initial collection of key-value pairs to populate the attributes.</param>
+        public Attributes(IEnumerable<KeyValuePair<string, object>> attributes)
+        {
+            _attributes = new List<KeyValuePair<string, object>>(attributes);
+        }
+
+        /// <summary>
+        /// Sets a key-value pair in the attributes collection.
+        /// </summary>
+        /// <param name="key">The key of the attribute.</param>
+        /// <param name="value">The value of the attribute.</param>
+        public void Set(string key, object value)
```

Comment:
Should this allow setting null?

Reply:
Update it to remove the existing item if the value is null.

Reply:
The spec wasn't very detailed and didn't contain any directions on the Attributes API other than set and get in addition to ` may contain additional methods`
I was trying to match how it is implemented in the ActivityTagsCollection https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivityTagsCollection.cs#L57, but we don't have to stick to what they have if we think it is a code smell, will update it.